src/Tasks/WriteCodeFragment.cs

```diff
@@ -221,12 +238,24 @@ private string GenerateCode(out string extension)
                         return null;
                     }
 
-                    attribute.Arguments.Add(orderedParameters[i]);
+                    providedOrderedParameters.Add(orderedParameters[i].Value);
                 }
 
-                foreach (CodeAttributeArgument namedParameter in namedParameters)
+                var attribute = new CodeAttributeDeclaration(new CodeTypeReference(attributeItem.ItemSpec));
+
+                // We might need the type of the attribute if we need to infer the
+                // types of the parameters. Search for it by the given type name,
+                // as well as within the namespaces that we automatically import.
+                Lazy<Type> attributeType = new(
+                    () => Type.GetType(attribute.Name) ?? NamespaceImports.Select(x => Type.GetType($"{x}.{attribute.Name}")).FirstOrDefault(),
```

Comment:
Could this throw with an invalid type?

Reply:
`Type.GetType(string)` will return null if the type is not found. There's an overload that takes a boolean, which lets you specify whether to throw an exception if the type is not found. If you'd prefer, I could change it to call that overload and pass false, which would make this more readable.

```csharp
Type.GetType(typeName, throwOnError: false)
```